File: src/variations/ScalaCardGame/Deck.scala
/**
  * Created by simmeringc on 1/22/18.
  */

package variations.ScalaCardGame

import scala.collection.mutable.ListBuffer

class Deck {
  var deckCards: ListBuffer[Card] = ListBuffer()
  var dealtCards: ListBuffer[Card] = ListBuffer()

  for (s <- Suits.values) {
    for (r <- Ranks.values) {
      deckCards += new Card(s, r)
    }
  }

  def dealTopCard(player: Player) = {
    val topCard = deckCards.head
    player.hand += topCard
    dealtCards += topCard
    deckCards = deckCards.drop(1)
  }

  def shuffleDeck() = {
    deckCards = scala.util.Random.shuffle(deckCards)
  }

  def printDeck() = {
    for (card <- deckCards) print(card.toString())
  }
}

File: src/variations/JavaCardGame/Deck.java
/**
 * Created by simmeringc on 1/22/18.
 */

package variations.JavaCardGame;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Deck {

  private List<Card> deckCards = new ArrayList<>();
  private List<Card> dealtCards = new ArrayList<>();

  public Deck() {
    for (Suits s : Suits.values()) {
      for (Ranks r : Ranks.values()) {
        Card c = new Card(s, r);
        this.deckCards.add(c);
      }
    }
  }

  public void dealTopCard(Player player) {
    Card card = deckCards.get(0);
    deckCards.remove(0);
    dealtCards.add(card);
    player.getHand().add(card);
  }

  // Storing cards in a Collection in order to
  // take advantage of Collections' built in shuffle method.
  public void shuffleDeck() {
    Collections.shuffle(this.deckCards);
  }

  @Override
  public String toString() {
    for (Card card : deckCards) {
      card.toString();
    }
    return null;
  }
}